### ISSUE TITLE
Default parameters not supported?

### ISSUE THREAD

**Issue by mkotsbak** [lihaoyi/autowire#42]
When I try interfaces like this:

trait Protocol {
  val dictionary: protocol.Dictionary
  val userRegistry: protocol.UserRegistry
}

trait UserRegistry {
  def createUser(newUser: User): User
  def lookupFirstNamePart(firstname: String, exact: Boolean = false): Seq[User]
}

and an implementation like this:

override def lookupFirstNamePart(firstnamePart: String, exact: Boolean = false)

the compile fails at line:

  val router = AutowireServer.route[Protocol](impl)

with error: "value lookupFirstNamePart$default$2 is not a member of io.widok.server.Controllers"

Am I doing anything wrong, or is the macro magic missing support for default values?

**Comment by tindzk** [comment]
I believe this is due to the default argument.

**Comment by mkotsbak** [comment]
Yes, me too, but there is no technical reasons it could not be supported? The default value should be inserted at server side, in case of not using Autowire on client side (third party client emulating the interface).

**Comment by mkotsbak** [comment]
Strangely I found this comment in the code indicating that it should have worked:

```
   * @param args Serialized arguments for the method that was called. Kept
   *             as a Map of arg-name -> serialized value. Values which
   *             exactly match the default value are omitted, and are
   *             simply re-constituted by the receiver.
```
